advanced/Scripts/chronometer.sh

```diff
@@ -8,98 +8,428 @@
+      dns_count="0"
+      [[ -n "${PIHOLE_DNS_2}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_3}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_4}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_5}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_6}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_7}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_8}" ]] && dns_count=$((dns_count+1))
+      [[ -n "${PIHOLE_DNS_9}" ]] && dns_count="$dns_count+"
+      ph_alts="${dns_count} others"
+    fi
+  fi
+  
+  sys_uptime=$(hrSecs "$(cut -d. -f1 /proc/uptime)")
+  sys_loadavg=$(cut -d " " -f1,2,3 /proc/loadavg)
+  
+  # Get CPU usage, only counting processes over 1% CPU as active
+  cpu_raw=$(ps -eo pcpu,rss --no-headers | grep -E -v "    0")
+  cpu_tasks=$(wc -l <<< "$cpu_raw")
+  cpu_taskact=$(sed -r "/(^ 0.)/d" <<< "$cpu_raw" | wc -l)
+  cpu_perc=$(awk '{sum+=$1} END {printf "%.0f\n", sum/'"$sys_cores"'}' <<< "$cpu_raw")
+  
+  # Get CPU clock speed
+  if [[ -n "$scaling_freq_file" ]]; then
+    cpu_mhz=$(( $(< /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq) / 1000 ))
+  else
+    cpu_mhz=$(lscpu | awk -F "[ .]+" '/MHz/ {print $4;exit}')
+  fi
+  
+  # Determine correct string format for CPU clock speed
+  if [[ -n "$cpu_mhz" ]]; then
+    [[ "$cpu_mhz" -le "999" ]] && cpu_freq="$cpu_mhz MHz" || cpu_freq="$(calcFunc "$cpu_mhz"/1000) Ghz"
+    [[ -n "$cpu_freq" ]] && cpu_freq_str=" @ $cpu_freq" || cpu_freq_str=""
+  fi
+  
+  # Determine colour for temperature
+  if [[ -n "$temp_file" ]]; then
+    if [[ "$temp_unit" == "C" ]]; then
+      cpu_temp=$(printf "%'.0fc\n" "$(calcFunc "$(< $temp_file) / 1000")")
+      
+      case "${cpu_temp::-1}" in
+        -*|[0-9]|[1-3][0-9]) cpu_col="$COL_LIGHT_BLUE";;
+        4[0-9]) cpu_col="";;
+        5[0-9]) cpu_col="$COL_YELLOW";;
+        6[0-9]) cpu_col="$COL_LIGHT_RED";;
+        *) cpu_col="$COL_URG_RED";;
+      esac
+      
+      # $COL_NC$COL_DARK_GRAY is needed for $COL_URG_RED
+      cpu_temp_str=", $cpu_col$cpu_temp$COL_NC$COL_DARK_GRAY"
+      
+    elif [[ "$temp_unit" == "F" ]]; then
+      cpu_temp=$(printf "%'.0ff\n" "$(calcFunc "($(< $temp_file) / 1000) * 9 / 5 + 32")")
+      
+      case "${cpu_temp::-1}" in
+        -*|[0-9]|[0-9][0-9]) cpu_col="$COL_LIGHT_BLUE";;
+        1[0-1][0-9]) cpu_col="";;
+        1[2-3][0-9]) cpu_col="$COL_YELLOW";;
+        1[4-5][0-9]) cpu_col="$COL_LIGHT_RED";;
+        *) cpu_col="$COL_URG_RED";;
+      esac
+      
+      cpu_temp_str=", $cpu_col$cpu_temp$COL_NC$COL_DARK_GRAY"
+      
+    else
+      cpu_temp_str=$(printf ", %'.0fk\n" "$(calcFunc "($(< $temp_file) / 1000) + 273.15")")
+    fi
+  else
+    cpu_temp_str=""
+  fi
+  
+  ram_raw=($(awk '/MemTotal:/{total=$2} /MemFree:/{free=$2} /Buffers:/{buffers=$2} /^Cached:/{cached=$2} END {printf "%.0f %.0f %.0f", (total-free-buffers-cached)*100/total, (total-free-buffers-cached)*1024, total*1024}' /proc/meminfo))
```

Comment:
It's part of the kernel, so since there is no `.` or `,` the number will always be the same, as for the tags, that would be a case to go in and `strings` the proper file... Not sure which though, I guess we could pull the linux kernel and check if need be.